### ISSUE TITLE
False positives with SCSS maps in color-named

### ISSUE THREAD

**Issue by ThePeach** [stylelint/stylelint#2009]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

This is a bug related to `color-named` and happens when a variable name has the same name of a key of a map

The needed SCSS snippet to trigger this is the following:

``` css
@function color($key) {
  @if map-has-key($colors, $key) {
    @return map-get($colors, $key);
  }
  return null;
}

// Color palette
$colors: (
  white: #FFFFFF,
  grey: #333333
);

$color-footer: color(white) !default;
```

> What stylelint configuration is needed to reproduce this issue?

``` json
{
  "rules": {
    "color-named": "never"
  }
}
```

> Which version of stylelint are you using?

`7.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

via `gulp-stylelint`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

"Yes, it's related to SCSS maps."

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

"The following warnings were flagged:"

``` shell
test.css
  10:3   ✖  Unexpected named color "white"    color-named
  11:3   ✖  Unexpected named color "grey"     color-named
  14:22  ✖  Unexpected named color "white"    color-named
```

**Comment by jeddy3** [comment]
@ThePeach Thanks for using the template.

You can the [`"ignore": ["inside-function"]`](https://github.com/stylelint/stylelint/tree/master/src/rules/color-named#ignore-inside-function) secondary option here e.g.

`"color-named": ["never", { "ignore": ["inside-function"] }]`

**Comment by ThePeach** [comment]
@jeddy3 thanks for the explanation. I don't have a use case on all the projects I'm using to do the automatic replacement with PostCSS and I'd rather write that once and forget about it.

I've used the suggested option and it seems to be working fine.
I would also agree that as a default behaviour it might be probably better, but -again- I don't have a use case for this instance.

**Comment by davidtheclark** [comment]
@jeddy3 I'm fine with ignoring maps by default.

**Comment by jeddy3** [comment]
@ThePeach Would you like to [contribute this fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-bugs)? It'll just be the case of [returning early if the "delc" isn't off standard syntax](https://github.com/stylelint/stylelint/blob/master/src/rules/declaration-colon-newline-after/index.js#L30).

**Comment by ThePeach** [comment]
@jeddy3 happy to help, I wonder if you linked to [declaration-colon-newline-after](https://github.com/stylelint/stylelint/blob/master/src/rules/declaration-colon-newline-after/index.js#L30) for a specific reason or am I missing something?

**Comment by jeddy3** [comment]
> I wonder if you linked to declaration-colon-newline-after for a specific reason or am I missing something?

It contains an example of how to return early if the "decl" isn't of standard syntax :)

**Comment by davidtheclark** [comment]
Closed by #2182.
